SpringKonstant/src/org/usfirst/frc6357/robotcode/subsystems/TFMini.java

```diff
@@ -63,38 +63,38 @@
     private int quality;     //Quality of signal
     private int checksum;    // byte 9: checksum
 
-    private boolean startRead;              // Boolean to start or stop reading values
+    private boolean startRead = false;              // Boolean to start or stop reading values
     public ReceivingState state;            // State of the packet receive
     public byte[] packet = new byte[9];     // Buffer for one packet
     private int count = 0;
 
+    private static TFMini m_instance = null;
+
     /**
      * Constructor, creates an instance of Serial Port class.
      * @param - SerialPort.Port enum, pick where it is plugged into
      */
-    public TFMini()
+    protected TFMini()
     {
         sensor = new SerialPort(baudRate, SerialPort.Port.kMXP, dataBits, SerialPort.Parity.kSpace, SerialPort.StopBits.kOne);
         state = ReceivingState.WAIT_START1;
         initialize();
     }
 
```

Comment:
Above and below this comment is the singleton design pattern. This class has a non-public constructor, which means it cannot be instantiated directly. Instead, the class holds a recursive static reference to itself and only instantiates that instance through a public static method. This method checks to see if the instance is null, if it is, it instantiates a new object. Then, the reference to the object is returned. This means that m_instance is assigned only once, but getInstance() can be called multiple times to get that one reference.